apps/server/src/modules/auth/next-auth.controller.ts

```diff
@@ -179,7 +179,7 @@ export class NextAuthController {
       });
       if (
         !req.headers?.referer ||
-        req.headers.referer.startsWith('https://accounts.google.com')
+        new URL(req.headers.referer).origin === 'https://accounts.google.com'
```

Comment:
The `new URL()` will throw an exception when the URL is invalid. Consider wrapping it in a try-catch block.

Reply:
Is it really possible that referer is invalid in the realworld?

Reply:
The "referer" is an HTTP header sent from the client which means it could be any value.